workflow/controller/controller.go

```diff
@@ -700,6 +707,17 @@ func (wfc *WorkflowController) newPodInformer() cache.SharedIndexInformer {
 			UpdateFunc: func(old, new interface{}) {
 				key, err := cache.MetaNamespaceKeyFunc(new)
 				if err == nil {
+					oldPod := old.(*apiv1.Pod)
+					newPod := new.(*apiv1.Pod)
+					if oldPod.GetResourceVersion() == newPod.GetResourceVersion() {
+						wfc.metrics.PodResourceVersionRepeated()
+						return
+					}
+					significant := oldPod.Status.Phase != newPod.Status.Phase
```

Comment:
we only update if the pod change is "significant" - phase is the very minimum this could be